framework/db/BaseActiveRecord.php

```diff
@@ -236,10 +236,9 @@ public function optimisticLock()
      */
     public function __get($name)
     {
-        if (isset($this->_attributes[$name]) || array_key_exists($name, $this->_attributes)) {
-            return $this->_attributes[$name];
-        } elseif ($this->hasAttribute($name)) {
-            return null;
+        // getAttribute has better performance than hasAttribute for attributes that are set.
+        if ((($value = $this->getAttribute($name)) !== null) || $this->hasAttribute($name)) {
+            return $value;
```

Comment:
Yes, that is explicitly desired.
Otherwise `$this->attribute = null;` will lead to undesired results. Basically we use `isset` first because it is faster, we could ommit it for readability.